controllers/nginx/pkg/config/config.go

```diff
@@ -307,20 +301,14 @@ func NewDefault() Configuration {
 	return cfg
 }
 
-// BuildLogFormatUpstream format the log_format upstream based on proxy_protocol
-func BuildLogFormatUpstream(useProxyProtocol bool, curLogFormatUpstream string) string {
-
-	// test if log_format comes from configmap
-	if curLogFormatUpstream != "" &&
-		curLogFormatUpstream != fmt.Sprintf(logFormatUpstream, "$proxy_protocol_addr") &&
-		curLogFormatUpstream != fmt.Sprintf(logFormatUpstream, "$remote_addr") {
-		return curLogFormatUpstream
-	}
-
-	if useProxyProtocol {
-		return fmt.Sprintf(logFormatUpstream, "$proxy_protocol_addr")
+// BuildLogFormatUpstream format the log_format upstream using
+// proxy_protocol_addr as remote client address if UseProxyProtocol
+// is enabled.
+func (cfg Configuration) BuildLogFormatUpstream() string {
```

Comment:
I mean, if I have this configmap 

```
apiVersion: v1
data:
  use-proxy-protocol: "true"
kind: ConfigMap
metadata:
  name: nginx-template
  namespace: default
```

and change to this?

```
apiVersion: v1
kind: ConfigMap
data:
  use-proxy-protocol: "true"
  log-format-upstream: $remote_user
metadata:
  name: nginx-template
  namespace: default
```

ingress will build a wrong log_format

```$cat /etc/nginx/nginx.conf |grep log_format
log_format upstreaminfo '$proxy_protocol_addr - $remote_user';
```